pool/src/lib.rs

```diff
@@ -810,16 +817,22 @@ impl LendingPool {
         assert_with_error!(env, amount <= balance, Error::NotEnoughAvailableUserBalance);
 
         let reserves = read_reserves(env);
-        // TODO: fix calc_account_data with balance after withdraw
-        let mb_account_data =
-            Self::calc_account_data(env, who, None, user_config, &reserves, false);
-        match mb_account_data {
-            Ok(account_data) => {
-                assert_with_error!(env, account_data.is_good_position(), Error::BadPosition)
+        if user_config.is_borrowing_any() {
+            let mb_account_data = Self::calc_account_data(
+                env,
+                who,
+                Some(s_token_after),
+                user_config,
+                &reserves,
+                false,
+            );
+            match mb_account_data {
```

Comment:
I see that we use it in a couple of places. Could you extract this block in a separate method?